os/net/mac/tsch/tsch.c

```diff
@@ -516,6 +516,12 @@ tsch_rx_process_pending()
       /* Pass to upper layers */
       packet_input();
     } else if(is_eb) {
+
+      /* Don't pass to upper layers, but still count it in link stats */
+      packetbuf_set_attr(PACKETBUF_ATTR_RSSI, current_input->rssi);
+      packetbuf_set_attr(PACKETBUF_ATTR_CHANNEL, current_input->channel);
+      link_stats_input_callback((const linkaddr_t *)frame.src_addr);
+
```

Comment:
Makes very much sense to me.

Maybe do a slight refactor, something like
```
   if(is_data) {
      /* Copy to packetbuf for processing, pass to upper layers */
      packetbuf_copyfrom(current_input->payload, current_input->len);
      packetbuf_set_attr(PACKETBUF_ATTR_RSSI, current_input->rssi);
      packetbuf_set_attr(PACKETBUF_ATTR_CHANNEL, current_input->channel);
      packet_input();
    }
    else if(is_eb) {
      /* Don't pass to upper layers, but still count it in link stats */
      packetbuf_set_attr(PACKETBUF_ATTR_RSSI, current_input->rssi);
      packetbuf_set_attr(PACKETBUF_ATTR_CHANNEL, current_input->channel);
      link_stats_input_callback((const linkaddr_t *)frame.src_addr);
      eb_input(current_input);
    }
```
or the `packetbuf_set_attr()`s in a `if(is_eb || is_data)` first

Reply:
I did some refactoring and updated the comments